src/tui/mod.rs

```diff
@@ -36,17 +36,17 @@ const SCROLL_AMOUNT: usize = 3;
 struct TuiState {
     quit: bool,
     scrollback: Vec<u8>,
-    line_count: usize,
-    scroll_offset: usize,
+    line_count: Saturating<usize>,
+    scroll_offset: Saturating<usize>,
 }
 
 impl Default for TuiState {
     fn default() -> Self {
         Self {
             quit: false,
             scrollback: Vec::with_capacity(TUI_SCROLLBACK_CAPACITY),
-            line_count: 1,
-            scroll_offset: 0,
+            line_count: Saturating(1),
```

Comment:
lol you could go even further and make this a `NonZeroUsize`

Reply:
`NonZeroUsize` isn't saturating, weirdly: https://doc.rust-lang.org/stable/core/num/struct.NonZeroUsize.html#method.saturating_add